xcresult/src/xcresult.rs

```diff
@@ -66,13 +69,19 @@ fn xcresulttool<T: AsRef<str>>(
 }
 
 impl XCResult {
-    pub fn new<T: AsRef<str>>(path: T) -> anyhow::Result<XCResult> {
+    pub fn new<T: AsRef<str>>(
+        path: T,
+        repo_url_parts: &RepoUrlParts,
+        org_url_slug: T,
```

Comment:
nit: reusing `T` here will necessitate that both `path` and `org_url_slug`. Conventionally, those will likely always be the same in this repo, but it's recommended to allow them to both be independent

```suggestion
    pub fn new<T: AsRef<str>, U: AsRef<str>>(
        path: T,
        repo_url_parts: &RepoUrlParts,
        org_url_slug: U,
```